irohad/torii/impl/command_service.cpp

```diff
@@ -207,7 +214,7 @@ namespace torii {
 
     log_->debug("StatusStream waiting finish, hash: {}", request_hash->hex());
 
-    if (not *finished) {
+    if (not*finished) {
```

Comment:
Remove shared_ptr wrapper as in patch 
[t.txt](https://github.com/hyperledger/iroha/files/2149468/t.txt)

Reply:
Ohh, please contact @luckychess first, it is here for a reason.

Reply:
It's here because lambda can live longer than main function. But as @l4l noticed that's an issue in a way how it's captured - it should be captured by value, not by reference.

Reply:
So, should I change the way it's being captured or it is issue for another P.R., and ToDo is to be left?

Reply:
It's better to fix it there, it only 1 line of change

Reply:
OK